### ISSUE TITLE
There is no pick action for Enchant decomposition equipment.

### ISSUE THREAD

**Issue by gamelgs** [vmangos/core#520]
There is no pick action for Enchant decomposition equipment.

The decomposed items are added to the backpack automatically.

**Client version**: 
1.12.1
**Commit hash**: 
ab11bde485814e2161bf8d174ff789035ecae2fe
**Operating system**: 
win10

**Comment by schell244** [comment]
I can confirm the issue, just disenchant any item. 

Mats will directly be added to players bag, loot ui with mats remains open, but can't be lootet. If player trys to loot, following error message is printet to log:

None (Guid: 0) is unsupported for looting. (Player XXX (Guid: XXX))

**Comment by schell244** [comment]
Issue is caused due to quickfix at SpellEffects.cpp:

```
void Spell::EffectDisEnchant(SpellEffectIndex /*eff_idx*/)
{
    if (m_caster->GetTypeId() != TYPEID_PLAYER)
        return;

    if (!itemTarget || !itemTarget->GetProto()->DisenchantID)
        return;

    Player* pCaster = static_cast<Player*>(m_caster);

    pCaster->UpdateCraftSkill(m_spellInfo->Id);
    pCaster->SendLoot(itemTarget->GetObjectGuid(), LOOT_DISENCHANTING);

    // quick solution to prevent exploiting; this will cause items loss on full bags though
    // in order to fix this correctly have to fix first plMover->SendLootRelease(lootGuid); @ MovementHandler
    // and in order to fix it the loot manager has to be rewritten; not me pls
    if (ObjectGuid lootGuid = pCaster->GetLootGuid())
        pCaster->GetSession()->DoLootRelease(lootGuid);

    // item will be removed at disenchanting end
}
```

**Comment by gamelgs** [comment]
Thank you for your reply, but I don't understand C + +. I just love this game and hope it will be better.

**Comment by ulfgebhardt** [comment]
I've seen this on other servers aswell - would you be interested in having a good look into this and report your findings @schell244 ?
-> Be aware that this has alot of implications regarding dupe/cheating.

**Comment by schell244** [comment]
I've done some simple in game tests a while ago in order to reproduce the mentioned exploid, but I could not find anything.
I will have another look on this issue and update this ticket if I find something. The person who wrote the hackfix mentioned _MovementHandler_ and  _loot manager has to be rewritten_ - if this is actually the case, than fixing this would currently be out of my c++ skill scope.

**Comment by schell244** [comment]
MovementHandler.cpp -> WorldSession::HandleMoverRelocation() 
HandleMoverRelocation() is always called when player moves. It's not called, when player is on transport and player just stands on it. It's called once after every loadingscreen (login, when player teleports, for example via hearthstone, ...)
Found this comment there:
```
        // super smart decision; rework required
        if (ObjectGuid lootGuid = pPlayerMover->GetLootGuid())
            if (!lootGuid.IsItem())
                pPlayerMover->SendLootRelease(lootGuid);
```
https://github.com/vmangos/core/blob/dfe94c8587788b049b8e4bedf93042bfcbe30678/src/game/Handlers/MovementHandler.cpp#L1050

**Comment by schell244** [comment]
fixed with f82e9b6 -  thanks @ratkosrb <3
